### DISCUSSION TITLE
How to send message from server to client?

### DISCUSSION THREAD

**Discussion by kof117** [arthuro555/THNK#59] [Q&A] [answered]
I'm migrating my game from P2P to THNK Rooms (Relay). I need to tell the player the data (stats and their player number), but I cannot figure out how to tell the client this. Is there not a way to do this, or am I missing something?

**Comment by arthuro555** [comment, marked as answer]
[As explained in the docs](https://thnk.cloud/docs/concepts/clients-and-servers/), the way communication between server and client works is that clients send messages to the server, the server ingests messages, and responds to all clients with new server state.

You can use a State variable to send any arbitrary data back to your clients. Since stats and player numbers are player specific, you'll want to [use a PlayerState variable](https://thnk.cloud/docs/concepts/state-variables/#player-state) for that state to be only sent to the players it concerns..

**Reply by kof117** [reply]
Great, thank you so much! I didn’t read the docs too closely, and I probably should’ve before I asked 😅
